### PR TITLE
[DO NOT MERGE] rustup: update to 1.16.0.

### LINKED ISSUE #5398: rustup broken
b079cb8f64452d4adf4a3faf45a58bf5595dac30 doesn't work as rustup as packaged
apparently doesn't create links in `.cargo/bin` in an clean environment:

```
$ rm -rf .rustup/ .cargo/bin/*
$ xbps-install rustup
$ sudo xbps-install rustup
Name   Action    Version           New version            Download size
rustup install   -                 1.15.0_1               - 
[...]
0 downloaded, 1 installed, 0 updated, 1 configured, 0 removed.
$ . /etc/profile.d/rustup.sh 
$ rustup default stable
[...]
info: downloading component 'cargo'
info: downloading component 'rust-docs'
info: installing component 'rustc'
info: installing component 'rust-std'
info: installing component 'cargo'
info: installing component 'rust-docs'
info: default toolchain set to 'stable-x86_64-unknown-linux-gnu'

  stable-x86_64-unknown-linux-gnu installed - rustc 1.30.1 (1433507eb 2018-11-07)

$ ls ~/.cargo/bin/
$ rustc
-bash: rustc: command not found
$ rustup which rustc
$HOME/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustc

```

cc @Cogitri

### PR DESCRIPTION
**Pull request by Cogitri**
fixes #5398

### CODE REVIEW

File: srcpkgs/rustup/INSTALL.msg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Please run
```
Comment (maxice8) [inline srcpkgs/rustup/INSTALL.msg]: can be done in one line.

Run rustup-init after initial install and updates.

**Comment by Cogitri** [comment]
This now carries a patch to symlink rustup @faulesocke

**Comment by faulesocke** [comment]
I've now had time to take a look at this. It works perfectly fine, really nice work! Can be merged.

**Comment by Cogitri** [comment]
Thanks for testing.

**Comment by Cogitri** [comment]
Mind taking a look at this @ebfe

**Comment by Cogitri** [comment]
Ah, oops, seems like I included an old version of the patch which doesn't _sym_, but _hardlink_ the /usr/bin/rustup-init to $CARGO_HOME/bin/rustup. Mind checking again (just rm rustup and run rustup-init again), @faulesocke? Nothing should effectively change though, other than us being able to update rustup with xbps :)

**Comment by faulesocke** [comment]
Oh, I thought this was intentional, but it's probably more clear if it's always a symlink (it was a symlink on my system anyway since /home and / are different filesystems). I'll check again once the compiler finishes ;)

**Comment by faulesocke** [comment]
Tested it, works fine as expected.

**Comment by Cogitri** [comment]
Alright, thanks for testing! :)

**Pull request merged**